### PR TITLE
Add KIC in Konnect with KGO example

### PR DESCRIPTION
**Pull request by mheap**
### Description

Updated KGO KIC guide to include Konnect

### Testing instructions

Preview link: [/gateway-operator/1.2.x/get-started/kic/create-gateway/](https://deploy-preview-7488--kongdocs.netlify.app/gateway-operator/1.2.x/get-started/kic/create-gateway/)

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

### CODE REVIEW

File: app/_src/gateway-operator/get-started/kic/create-gateway.md [hunk 1]
```diff
@@ -17,11 +17,127 @@ chapter: 2
 > in order to get your `Gateway` up and running.
 {% endif_version %}
 
-To use the Gateway API resources to configure your routes, you need to create a `GatewayClass` instance and create a `Gateway` resource that listens on the ports that you need.
+Creating `GatewayClass` and `Gateway` resources in Kubernetes causes {{ site.kgo_product_name }} to create a {{ site.kic_product_name }} and {{ site.base_gateway }} deployment. 
+
+## GatewayConfiguration
```
[the remark this answers is not in the document]
Comment (cloudjumpercat) [reply app/_src/gateway-operator/get-started/kic/create-gateway.md:22]: I also had to have my minikube tunnel running at the same time or I got errors, is that something we should mention here too?

File: app/_src/gateway-operator/get-started/kic/create-gateway.md [hunk 2]
```diff
@@ -17,11 +17,127 @@ chapter: 2
...
+
+You can customize your {{ site.kic_product_name }} and {{ site.base_gateway }} deployments using the `GatewayConfiguration` CRD. This allows you to control the image being used, and set any environment variables required.
```
Comment (cloudjumpercat) [inline app/_src/gateway-operator/get-started/kic/create-gateway.md:24]:
```suggestion
You can customize your {{ site.kic_product_name }} and {{ site.base_gateway }} deployments using the `GatewayConfiguration` CRD. This allows you to control the image being used, and set any required environment variables.
```

File: app/_src/gateway-operator/get-started/kic/create-gateway.md [hunk 3]
```diff
@@ -17,11 +17,127 @@ chapter: 2
...
+{%- if_version gte:1.2.x %}
+ If you are creating a KIC in Konnect deployment, you need to customize the deployment to contain your control plane ID and authentication certificate
```
Comment (cloudjumpercat) [inline app/_src/gateway-operator/get-started/kic/create-gateway.md:26]:
```suggestion
 If you are creating a KIC in {{site.konnect_short_name}} deployment, you need to customize the deployment to contain your control plane ID and authentication certificate.
```

File: app/_src/gateway-operator/get-started/kic/create-gateway.md [hunk 4]
```diff
@@ -17,11 +17,127 @@ chapter: 2
...
+          - name: konnect-client-tls
+            secret:
+              secretName: konnect-client-tls
+              defaultMode: 420' | kubectl apply -f -
+```
+{% endnavtab %}
+{% endif_version %}
+{% navtab On-Prem %}
+
+This example shows how to customize the log level of {{ site.kic_product_name }}:
```
Comment (cloudjumpercat) [inline app/_src/gateway-operator/get-started/kic/create-gateway.md:137]:
```suggestion
Use the following example to customize the log level of {{ site.kic_product_name }}:
```

File: app/_src/gateway-operator/get-started/kic/create-gateway.md [hunk 5]
```diff
@@ -46,8 +162,23 @@ spec:
...
+
+The results should look like this:
+
+```text
+gatewayconfiguration.gateway-operator.konghq.com/kong created
+```
+
+## GatewayClass
+
+To use the Gateway API resources to configure your routes, you need to create a `GatewayClass` instance and create a `Gateway` resource that listens on the ports that you need.
```
Comment (cloudjumpercat) [inline app/_src/gateway-operator/get-started/kic/create-gateway.md:178]: Does this only apply to on prem?

File: app/_src/gateway-operator/get-started/kic/create-gateway.md [hunk 6]
```diff
@@ -70,15 +201,12 @@ spec:
...
-
-' | kubectl apply -f -
+    port: 80' | kubectl apply -f -
 ```
 
 The results should look like this:
 
 ```text
-gatewayconfiguration.gateway-operator.konghq.com/kong created
 gatewayclass.gateway.networking.k8s.io/kong created
```
Comment (cloudjumpercat) [inline app/_src/gateway-operator/get-started/kic/create-gateway.md:210]: For lines 214-218, I ran that and it worked, but I don't understand why I needed that or what I'm doing/using the variable `PROXY_IP` for.

**Review by cloudjumpercat — COMMENTED** [review]
@mheap Tested the Konnect and on prem workflows and left a few suggestions/comments/questions!

File: app/_src/gateway-operator/get-started/kic/create-gateway.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mheap) [reply app/_src/gateway-operator/get-started/kic/create-gateway.md:22]: minikube tunnel is an implementation detail. We used to have this kind of thing in the docs, but decided that people should know how to use their k8s tools

File: app/_src/gateway-operator/get-started/kic/create-gateway.md [hunk 7]
```diff
@@ -17,11 +17,127 @@ chapter: 2
...
+1. Click **Generate Certificate** in step 3.
+1. Save the contents of **Cluster Certificate** in a file named `tls.crt`. Save the contents of **Cluster Key** in a file named `tls.key`.
+1. Create a Kubernetes secret containing the cluster certificate:
+
+    ```bash
+    kubectl create secret tls konnect-client-tls --cert=/{PATH_TO_FILE}/tls.crt --key=/{PATH_TO_FILE}/tls.key
+    ```
+1. In the **Configuration parameters** step 4, find the value of `runtimeGroupID`. Replace `YOUR_CP_ID` with the control plane ID in the following manifest.
+1. In the **Configuration parameters** step 4, find the value of `cluster_telemetry_endpoint`. The first segment of that value is the control plane endpoint for your cluster. For example, if the value of `cluster_telemetry_endpoint` is `36fc5d01be.us.cp0.konghq.com`, then the control plane endpoint of the cluster is `36fc5d01be`. Replace `YOUR_CP_ENDPOINT` with your control plane ID in the following manifest.
+1. Deploy the data plane with `kubectl apply`:
```
[the remark this answers is not in the document]
Comment (mheap) [reply app/_src/gateway-operator/get-started/kic/create-gateway.md:48]: No, the Operator replaces `helm install`

File: app/_src/gateway-operator/get-started/kic/create-gateway.md (hunk 5 above)
Comment (mheap) [reply app/_src/gateway-operator/get-started/kic/create-gateway.md:178]: This applies to on prem and Konnect

File: app/_src/gateway-operator/get-started/kic/create-gateway.md (hunk 6 above)
Comment (mheap) [reply app/_src/gateway-operator/get-started/kic/create-gateway.md:210]: Good call out. I've moved this to the "Add a route" page

**Review by Guaris — APPROVED** [review]

**Comment by pmalek** [comment]
I've made small changes to the guide (adapting to UI changes and such).

1 thing that is making me pause is the fact that we're adding a Konnect focused guide in  KIC focused docs subtree (not sure about nomenclature here).

image

While this might be slightly confusing to readers I'm not sure if we have a better option for this. 

One thing I'd consider is to split the 2 pages into 3:

- Konnect DataPlanes
- Konnect KIC 
- On Prem KIC

That might be done separately and this guide is 👍 to me.

**Pull request merged**